embeddings/pipeline/preprocessing_pipeline.py

```diff
@@ -52,120 +58,134 @@ def run(self) -> TransformationResult:
         return result
 
 
-class FlairTextClassificationPreprocessingPipeline(
-    PreprocessingPipeline[str, datasets.DatasetDict, Corpus]
+# Mypy currently properly don't handle dataclasses with abstract methods  https://github.com/python/mypy/issues/5374
+@dataclass  # type: ignore
+class FlairPreprocessingPipeline(
+    PreprocessingPipeline[str, Union[datasets.DatasetDict, Corpus], Corpus],
+    abc.ABC,
 ):
-    def __init__(
+    persister: Union[FlairConllPersister[Corpus], FlairPicklePersister[Corpus, Corpus]] = field(
+        init=False
+    )
+
+    dataset_name_or_path: str
+    persist_path: str
+    input_column_name: Union[str, Tuple[str, str]]
+    target_column_name: str
+    datasets_path: Path = DATASET_PATH
+    sample_missing_splits: Optional[Tuple[Optional[float], Optional[float]]] = None
+    downsample_splits: Optional[Tuple[Optional[float], Optional[float], Optional[float]]] = None
+    downsample_splits_stratification: bool = True
```

Comment:
how about prefixing boolean variables with should_ or if_ etc?

Reply:
We could discuss it, but I think I would move it to our general refactor issue

Reply:
Added to #120